File: pascal/CaveEscape2.pas
program GameMain;
uses SplashKit, sysUtils;

function GetNewPlayer(): Sprite;
begin
  result := CreateSprite(BitmapNamed('Player'), AnimationScriptNamed('PlayerAnimations'));
  SpriteSetX(result, ScreenWidth() / 2 - SpriteWidth(result));
  SpriteSetY(result, ScreenHeight() / 2);
  SpriteStartAnimation(result, 'Fly');
end;

procedure Main();
var
  player: Sprite;
begin
  OpenWindow('Cave Escape', 432, 768);
  LoadResourceBundle('CaveEscape', 'CaveEscape.txt');

  player := GetNewPlayer();

  repeat // The game loop...
    ProcessEvents();
    ClearScreen(ColorWhite);
    UpdateSprite(player);
    DrawSprite(player);
    RefreshScreen(60);
  until WindowCloseRequested('Cave Escape');
end;

begin
  Main();
end.

File: c/cave_escape_2.cpp
#include <stdio.h>
#include "splashkit.h"

sprite get_new_player()
{
  sprite result;
  result = create_sprite(bitmap_named("Player"), animation_script_named("PlayerAnimations"));
  sprite_set_x(result, screen_width() / 2 - sprite_width(result));
  sprite_set_y(result, screen_height() / 2);
  sprite_start_animation(result, "Fly");0)

  return result;
}

int main()
{
    sprite player;

    open_window("Cave Escape", 432, 768);
    load_resource_bundle("CaveEscape", "CaveEscape.txt");

    player = get_new_player();

    do // The game loop...
    {
      process_events();
      clear_screen(COLOR_WHITE);
      update_sprite(player);
      draw_sprite(player);
      refresh_screen(60);
    } while(!window_close_requested("Cave Escape"));

    return 0;
}
